### PR TITLE
test: Add a new Github action to track failing conformance tests

### PR DESCRIPTION
**Pull request by danieljbruce**
**Summary:**

This PR adds a required Github action that ensures all conformance tests pass except for the conformance tests listed in the `known_failures.txt` file. This way we can be sure that changes won't be made in the future that cause conformance tests to fail that were passing before without us knowing.

**Background:**

[This "conformance test PR"](https://github.com/googleapis/nodejs-bigtable/pull/1349) added a Github action that checks to see if all conformance tests are passing. This Github action fails even if one conformance test fails, but many of the conformance tests are failing at this time. So this existing Github action is useful for tracking the overall status of the conformance tests, but we need another Github action to ensure that addressing a known failure doesn't cause other conformance tests to fail.

**Changes:**

`.github/sync-repo-settings.yaml`: The new Github action called `mandatory-conformance` is added as a required status check in the continuous integration pipeline. This check makes it so we cannot merge PRs when a code change causes a passing conformance test to suddenly begin failing without adding that test to the `known_failures.txt` list. This will ensure that we do not merge PRs that introduce failing conformance tests without being explicit about introducing the test failure.

`.github/workflows/mandatory-conformance.yaml`: This file was created by copying `conformance.yaml` from [this "conformance test PR"](https://github.com/googleapis/nodejs-bigtable/pull/1349) and then changing every mention of `conformance` to `mandatory-conformance` to introduce a new Github action that must pass in order for the PR to be merged.

`.kokoro/mandatory-conformance.sh`: This file was created by copying `conformance.sh` from [this "conformance test PR"](https://github.com/googleapis/nodejs-bigtable/pull/1349) and then changing the line of code that runs the conformance tests to only run tests that are not in the `known_failures.txt` list. The line of code that prints the logs is also omitted so that it doesn't overwrite the logs produced in `conformance.sh`.

`known_failures.txt`: In this file a backslash is added to address an error and each test is now listed on a separate line so that each time a PR addresses a specific conformance test the diff will show us exactly what test has been addressed. If the test names are not on separate lines then it may be challenging to see exactly what tests have been added or removed to the known failures list. Also, this Github action revealed a list of other conformance tests that are failing that have been added to the list including:
TestReadRows_Retry_StreamReset
TestCheckAndMutateRow_Generic_Headers
TestMutateRows_Generic_DeadlineExceeded
TestReadRows_Generic_CloseClient
TestReadRows_Generic_DeadlineExceeded
TestReadRows_Retry_WithRoutingCookie
TestReadRows_Retry_WithRoutingCookie_MultipleErrorResponses
TestReadRows_Retry_WithRetryInfo
TestReadRows_Retry_WithRetryInfo_MultipleErrorResponse
TestSampleRowKeys_Generic_Headers
TestSampleRowKeys_Generic_DeadlineExceeded
TestSampleRowKeys_Retry_WithRoutingCookie

**Next Steps:**

In future PRs we will address one conformance test at a time shortening the `known_failures.txt` list with each conformance test we successful change from failing to passing.

### CODE REVIEW

File: .github/sync-repo-settings.yaml [hunk 1]
```diff
@@ -8,6 +8,7 @@ branchProtectionRules:
       - "ci/kokoro: Samples test"
       - "ci/kokoro: System test"
       - lint
+      - mandatory-conformance
```
Comment (danieljbruce) [inline .github/sync-repo-settings.yaml:11]: This is the new required Github action

File: .github/workflows/mandatory-conformance.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+# Github action job to test core java library features on
+# downstream client libraries before they are released.
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+name: mandatory-conformance
+jobs:
+  mandatory-conformance:
```
Comment (danieljbruce) [inline .github/workflows/mandatory-conformance.yaml:23]: This file is just like [the changes in this PR](https://github.com/googleapis/nodejs-bigtable/pull/1349/files#diff-7bfd467904d753bc34e8f91ffbb8165dc048bbe3610af168c600072d0ec473c0R23), but with `conformance` renamed to `mandatory-conformance`.

File: .kokoro/mandatory-conformance.sh [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+# Build and start the proxy in a separate process
+pushd .
+npm install
+nohup npm run testproxy &
+proxyPID=$!
+popd
+
+# Run the conformance test skipping known failures
+cd cloud-bigtable-clients-test/tests
+eval "go test -v -proxy_addr=:9999 -skip `tr -d '\n' < ../../testproxy/known_failures.txt`"
```
Comment (danieljbruce) [inline .kokoro/mandatory-conformance.sh:41]: This file is an exact copy of `conformance.sh` only with [this line](https://github.com/googleapis/nodejs-bigtable/pull/1349/files#diff-08b007a16e10891ada78edba8173c7323e74500fa6e1600c5e1dc7a721df5388R41) adjusted to not test against known failures and another line of code removed that prints logs.

File: testproxy/known_failures.txt [hunk 4]
```diff
@@ -1 +1,29 @@
-TestMutateRow_Generic_Headers\|TestMutateRow_Generic_DeadlineExceeded|TestMutateRows_Generic_CloseClient\|TestMutateRows_Retry_WithRoutingCookie\|TestReadModifyWriteRow_Generic_Headers\|TestReadModifyWriteRow_NoRetry_TransientError\|TestReadModifyWriteRow_Generic_DeadlineExceeded\|TestReadRow_Generic_DeadlineExceeded\|TestReadRow_Retry_WithRoutingCookie\|TestReadRow_Retry_WithRetryInfo\|TestReadRows_ReverseScans_FeatureFlag_Enabled\|TestReadRows_NoRetry_OutOfOrderError_Reverse\|TestReadRows_Retry_PausedScan\|TestReadRows_Retry_LastScannedRow\|TestReadRows_Retry_LastScannedRow_Reverse\|TestCheckAndMutateRow_NoRetry_TransientError\|TestCheckAndMutateRow_Generic_DeadlineExceeded
```
Comment (danieljbruce) [inline testproxy/known_failures.txt:1]: Changes:
1. Backslash after `TestMutateRow_Generic_DeadlineExceeded`
2. All test names on their own line
3. Lots of tests added to the file that are actually failing

File: .kokoro/mandatory-conformance.sh (hunk 3 above)
Comment (kevkim-codes) [reply .kokoro/mandatory-conformance.sh:41]: Perfect, thanks for doing this.

**Review by kevkim-codes — APPROVED** [review]

**Pull request merged**